mongodb.py

```diff
@@ -156,53 +156,76 @@ def do_server_status(self):
                 pass
 
         # uptime
-        self.submit('gauge', 'uptime', server_status['uptime'])
+        if 'uptime' in server_status:
+            self.submit('gauge', 'uptime',
+                        server_status['uptime'])
 
         # operations
-        if 'opcounters' in server_status:
-            for k, v in server_status['opcounters'].items():
-                self.submit('counter', 'opcounters.' + k, v)
+        opcounters_obj = server_status.get('opcounters')
+        if opcounters_obj:
+            for counter_key in opcounters_obj.keys():
+                self.submit('counter', 'opcounters.{}'.format(counter_key),
+                            opcounters_obj[counter_key])
 
         # memory
-        if 'mem' in server_status:
+        mem_obj = server_status.get('mem')
+        if mem_obj:
             for t in ['resident', 'virtual', 'mapped']:
-                self.submit('gauge', 'mem.' + t, server_status['mem'][t])
+                if t not in mem_obj:
+                    continue
+
+                self.submit('gauge', 'mem.{}'.format(t),
+                            mem_obj[t])
 
         # network
-        if 'network' in server_status:
+        network_obj = server_status.get('network')
+        if network_obj:
             for t in ['bytesIn', 'bytesOut', 'numRequests']:
-                self.submit('counter', 'network.' + t,
-                            server_status['network'][t])
+                if t not in network_obj:
+                    continue
+
+                self.submit('counter', 'network.{}'.format(t),
+                            network_obj[t])
 
         # connections
-        if 'connections' in server_status:
+        connections_obj = server_status.get('connections')
+        if connections_obj:
             for t in ['current', 'available', 'totalCreated']:
-                self.submit('gauge', 'connections.' + t,
-                            server_status['connections'][t])
+                if t not in connections_obj:
+                    continue
+
+                self.submit('gauge', 'connections.{}'.format(t),
+                            connections_obj[t])
 
         # background flush
-        if 'backgroundFlushing' in server_status:
-            self.submit('counter', 'backgroundFlushing.flushes',
-                        server_status['backgroundFlushing']['flushes'])
-            self.submit('gauge', 'backgroundFlushing.average_ms',
-                        server_status['backgroundFlushing']['average_ms'])
-            self.submit('gauge', 'backgroundFlushing.last_ms',
-                        server_status['backgroundFlushing']['last_ms'])
+        backgroundflushing_obj = server_status.get('backgroundFlushing')
+        if backgroundflushing_obj:
+            for t in ['flushes', 'average_ms', 'last_ms']:
+                if t not in backgroundflushing_obj:
+                    continue
+
+                self.submit('counter', 'backgroundFlushing.{}'.format(t),
```

Comment:
The second and third metrics were gauges before.

Reply:
That was my mistake. That was an error in copy/paste.